hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/ReplicaMap.java

```diff
@@ -108,16 +109,34 @@ ReplicaInfo get(String bpid, long blockId) {
     checkBlockPool(bpid);
     try (AutoCloseableLock l = lock.acquire()) {
       // check inner-maps; each of them will have their own synchronization.
-      for (VolumeReplicaMap inner : innerReplicaMaps.values()) {
-        ReplicaInfo info = inner.get(bpid, blockId);
-        if (info != null) {
-          return info;
+      for (FsVolumeImpl fsVolume : innerReplicaMaps.keySet()) {
```

Comment:
This change is not related to the bug mentioned in the description, right?

Reply:
Yes, this is not related. It is part of a separate commit to prioritize searching in local volumes since most blocks are expected to be local.